docs/csharp/tutorials/default-interface-members-versions.md

```diff
@@ -0,0 +1,80 @@
+
+## Scenario overview
+
+This tutorial starts with version 1 of a customer relationship library. The company that built this library intended customers with existing applications to adopt their library. They provided minimal interface definitions for users of their library to implement. Here's the interface definition for a customer:
+
+[!code-csharp[InitialCustomerInterface](~/samples/csharp/tutorials/default-interface-members-versions/starter/customer-relationship/ICustomer.cs?name=SnippetICustomerVersion1)]
+
+They defined a second interface that represents an order:
+
+[!code-csharp[InitialOrderInterface](~/samples/csharp/tutorials/default-interface-members-versions/starter/customer-relationship/IOrder.cs?name=SnippetIorderVersion1)]
+
+From those interfaces, the team could build a library for their users to create a better experience for their customers. Their goal was to create a deeper relationship with existing customers and improve their relationships with new customers.
+
+Now, it's time to upgrade the library for the next release. One of the requested features enables a loyalty discount for customers that have lots of orders. This new loyalty discount gets applied whenever a customer makes an order. The specific discount is a property of each individual customer. Each implementation of ICustomer can set different rules for the loyalty discount. 
+
+The most natural way to add this functionality is to enhance the `ICustomer` interface with a method to apply any loyalty discount. This design suggestion caused concern among experienced developers: "Interfaces are immutable once they've been released! This is a breaking change!" C# 8.0 adds *default interface implementations* for upgrading interfaces. The library authors can add new members to the interface and provide a default implementation for those members.
+
+Default interface implementations enable developers to upgrade an interface while still enabling any implementors to override that implementation. Users of the library can accept the default implementation as a non-breaking change. If their business rules are different, they can override.
+
+## Upgrade with default interface members
+
+The team agreed on the most likely default implementation: a loyalty discount for customers 
+
+The upgrade should provide the functionality to set two properties: the number of orders needed to be eligible for the discount, and the percentage of the discount. This makes it a perfect scenario for default interface members. You can add a method to the ICustomer interface, and provide the most likely implementation. All existing, and any new implementations can use the default implementation, or provide their own.
```

Comment:
I feel like in this scenario the default implementation would be to provide 0 discount. This feature is described as an opt-in feature where users can provide loyalty discounts. I can see where the creators of the library provide the default implementation that does nothing, so as to not break compatibility and enable users to provide their own discount logic. So the first code example of the implementation would end up being `public decimal ComputeLoyaltyDiscount() => 0;`

Reply:
I think I'll leave this as is. It flows easier into the next sections. You are right, that a reasonable default is "don't make any changes". That would make the scenario less interesting.

Reply:
I do want to leave the scenario the same. It flows better into the additional features. (But you do have a point)